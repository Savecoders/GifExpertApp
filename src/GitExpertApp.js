import { useState } from 'react';
import AddCategory from './components/AddCategory';
import GifGrid from './components/GifGrid';
import NavBar from './components/NavBar';
const GitExpertApp = () => {
	/* const categories = [
		'One Piece',
		'Hunter X Hunter',
		'Full Metal Alchemist',
		'One Punch',
	]; */
	const [categories, setCategories] = useState(['Full Metal Alchemists']);
	/* const handleAdd = () => setCategories((cats) => [...cats, 'Boku no hero']);
	const handleDelete = () => setCategories((cats) => cats.splice(0, cats.length - 1)); */
	/* 	categories.filter((categorie) => categorie !== categories[categories.length - 1]) */
	return (
		<>
			<NavBar title={'GifExpertApp'} />
			<AddCategory setCategories={setCategories} />
			{/*The keys are an important part for reactj to take into account
         the state of our application and thus have a specific order and
        if any change was made, react will take it into account */}
			{categories.map((category) => (
				<GifGrid key={category} category={category} />
			))}
		</>
	);
};

export default GitExpertApp;
