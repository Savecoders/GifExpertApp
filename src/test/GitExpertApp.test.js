import { shallow } from 'enzyme';
import GitExpertApp from '../GitExpertApp';

describe('render <GifExpertApp/>', () => {
	test('shallow components', () => {
		const wrapper = shallow(<GitExpertApp />);
		expect(wrapper).toMatchSnapshot();
	});
	test('have categories', () => {
		const categories = ['One Puch', 'Hunter X Hunters'];
		const wrapper = shallow(<GitExpertApp defaultCategoies={categories} />);
		expect(wrapper).toMatchSnapshot();
		expect(wrapper.find('GifGrid').length).toBe(categories.length);
	});
});
