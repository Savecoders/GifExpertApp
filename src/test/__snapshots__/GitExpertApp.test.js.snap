// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render <GifExpertApp/> have categories 1`] = `
<Fragment>
  <NavBar
    title="GifExpertApp"
  />
  <AddCategory
    setCategories={[Function]}
  />
  <GifGrid
    category="One Puch"
    key="One Puch"
  />
  <GifGrid
    category="Hunter X Hunters"
    key="Hunter X Hunters"
  />
</Fragment>
`;

exports[`render <GifExpertApp/> shallow components 1`] = `
<Fragment>
  <NavBar
    title="GifExpertApp"
  />
  <AddCategory
    setCategories={[Function]}
  />
</Fragment>
`;
